git-link.el

```diff
@@ -191,7 +199,10 @@ As an example, \"gitlab\" will match with both \"gitlab.com\" and
   (car (git-link--exec "symbolic-ref" "--short" "HEAD")))
 
 (defun git-link--repo-root ()
-  (car (git-link--exec "rev-parse" "--show-toplevel")))
+  (let ((dir (car (git-link--exec "rev-parse" "--show-toplevel"))))
+    (if (file-remote-p default-directory)
```

Comment:
Hi, the problem with this is that one can run `M-x cd` and change the default directory. Buffer remains as is, but repo root returned by this will be wrong.

Reply:
> Hi, the problem with this is that one can run M-x cd and change the default directory. Buffer remains as is, but repo root returned by this will be wrong.

Yes, but this case also fails on non-tramp buffers, because `git-link--exec` executes in `default-directory` so `git-link--repo-root` fails to get the correct root.

IMO this is a edge case no user will unintentional encounter.

Reply:
>  git-link--exec executes in default-directory

Hurmmm, yes. Ok, will try to somehow address this later then. 

Thanks again for this!